src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -269,6 +273,74 @@ def __bool__(self) -> bool:
         return bool(self.req_strings)
 
 
+@dataclass(frozen=True)
+class ResolvePexConfig:
+    """Configuration from `[python]` that impacts how the resolve is created."""
+
+    constraints_file: tuple[Digest, FileEntry] | None
```

Comment:
This will soon also have `--only-binary` and `--no-binary`. Possibly `[python-repos]` options, pending Slack discussion